src/jit/hwintrinsicxarch.cpp

```diff
@@ -218,6 +228,39 @@ int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic)
     return hwIntrinsicInfoArray[intrinsic - NI_HW_INTRINSIC_START - 1].numArgs;
 }
 
+//------------------------------------------------------------------------
+// numArgsOfHWIntrinsic: get the number of arguments based on baseType
+//
+// Arguments:
+//    intrinsic -- id of the intrinsic function.
+//    baseType  -- var_types type of intrinsic
+//
+// Return Value:
+//     number of arguments
+//
+int Compiler::numArgsOfHWIntrinsic(NamedIntrinsic intrinsic, var_types baseType)
```

Comment:
As I said previously, this function should be more general that does not rely on `baseType` nor check intrinsic ID.

Reply:
@fiigii @4creators  - I agree that it would be more efficient to simply look at the children of the node to get the argument count, but I don't think we need the extra flag, since for the case with a fixed number of args that requires checking both the flag and the count, instead of simply getting the `numArgs` field.

Reply:
@CarolEidt Thank you for the comment.